rust/common/cookieless/src/salt_cache.rs

```diff
@@ -0,0 +1,354 @@
+use base64::{engine::general_purpose, Engine};
```

Comment:
This is a pretty direct port of the same function in TS

This is why e.g. I didn't want to use a struct for YYYYMMDD, I was the code as similar as possible